src/Microsoft.OData.Core/Batch/ODataBatchWriter.cs

```diff
@@ -573,11 +571,91 @@ protected void SetState(BatchWriterState newState)
         protected ODataBatchOperationResponseMessage BuildOperationResponseMessage(Stream outputStream,
             string contentId, string groupId)
         {
-            Func<Stream> streamCreatorFunc = () => ODataBatchUtils.CreateBatchOperationWriteStream(outputStream, this);
+            Func<Stream> streamCreatorFunc = () => ODataBatchUtils.CreateBatchOperationWriteStream(
+                outputStream, this, this.outputContext.Synchronous);
             return new ODataBatchOperationResponseMessage(streamCreatorFunc, /*headers*/ null, this, contentId,
                 this.payloadUriConverter.BatchMessagePayloadUriConverter, /*writing*/ true, this.container, groupId);
         }
 
+        /// <summary>
+        /// Asnchronously starts a new batch.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        protected virtual Task WriteStartBatchImplementationAsync()
+        {
+            return TaskUtils.GetTaskForSynchronousOperation(this.WriteStartBatchImplementation);
+        }
+
+        /// <summary>
+        /// Asynchronously ends a batch.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        protected virtual Task WriteEndBatchImplementationAsync()

```

Comment:
Because this method as defined here is a pass-through.
`GetTaskForSynchronousOperation` method defined in `TaskUtils`. The way it is implemented is such that it executes the synchronous operation in a try...catch and returns a completed task if the execution is successful and a faulted task if not (_unless the exception is not "catchable"_).
The fact the `GetTaskForSynchronousOperation` returns a task and we are doing nothing else within `WriteEndBatchImplementationAsync` apart from calling the method means its safe to elide async/await in this particular case.